File: backend/src/WhereIsMyMoney.Api/Models/EnableBankingModels/EnableBankingBankSession.cs
namespace WhereIsMyMoney.Api.Models.EnableBankingModels;

public class EnableBankingBankSession
{
    public long Id { get; set; }
    public long IntegrationId { get; set; }
    public long AccountId { get; set; }
    public string SessionId { get; set; } = string.Empty;
    public string AspspName { get; set; } = string.Empty;
    public string AspspCountry { get; set; } = string.Empty;
    public DateTime ValidUntil { get; set; }
    public string AccountsJson { get; set; } = "[]"; // JSON array of account UIDs
    public DateTime CreatedAtUtc { get; set; }
}

File: frontend/WhereIsMyMoneyUI/src/app/models/import/EnableBankingBankSession.ts
export interface EnableBankingBankSession {
  id: number;
  integrationId: number;
  accountId: number;
  sessionId: string;
  aspspName: string;
  aspspCountry: string;
  validUntil: string;
  accountsJson: string;
  createdAtUtc: string;
}
